sdcflows/interfaces/fmap.py

```diff
@@ -666,17 +666,21 @@ def au2rads(in_file, newpath=None):
     data = im.get_fdata(dtype='float32')
     hdr = im.header.copy()
 
-    # First center data around 0.0.
-    data -= mode(data, axis=None)[0][0]
+    if np.allclose((data.min(), data.max()), (-np.pi, np.pi), atol=0.01):
+        # Already in rads, but wrap to 0 - 2pi
+        data[data < 0] += 2 * np.pi
```

Comment:
Why only negatives?
```suggestion
        data += data.min()
```

Reply:
Because this function sets the range to (0, 2pi), not (pi, 3pi). And it seems unwise to go precisely counter-phase by adding `np.pi` to everything.

Reply:
Sorry, I had the suggestion above sign-flipped. I've fixed it.

This is going to wrap and scale only negatives - you don't want either (and even less only for the lower half of the histogram).

If I understand correctly, the initial range goes from -pi to pi. Adding pi will move the range to 0-2pi

Reply:
So does fieldmap correction treat (θ±π) the same? Shifting by half phase feels problematic.

Reply:
To be clear, this is equivalent to:

```
data = (data + 2 * np.pi) % (2 * np.pi)
```

But without bothering add, divide and round things in the (0, pi) range. It doesn't feel controversial to me. `data += np.pi` does.

Reply:
> So does fieldmap correction treat (θ±π) the same? Shifting by half phase feels problematic.

FSL FUGUE theoretically only unwraps phase maps in the (0, 2π) range. It seems to work with (-π, π), though.

I don't know why `data -= np.pi` is controversial - it just moves all values from (-π, π) to (0, 2π).

The transformation you propose will wrap negative phase values (which you definitely don't want to do).

Normally, a phase maps' histogram looks like a centered pyramid. Doing this, we are selecting the left side of the pyramid, shifting it to overlap the right side of the pyramid and finally checking no values have gone over 2pi. That will introduce another wrapping that FUGUE might or might not pick up.

Reply:
Let's talk in the meeting.